File: xijinhui_order/app/Models/FoodDish.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Facades\Storage;

class FoodDish extends Model
{
    use HasFactory, SoftDeletes;

    protected $table = 'food_dishes';
    protected $fillable = [
        'category_id',
        'food_code',
        'english_name',
        'chinese_name',
        'khmer_name',
        'description',
        'food_price',
        'remark',
        'food_photo',
        'english_keyword',
        'chinese_keyword',
        'thai_keyword',
        'display_order',

        'is_enable',
        'is_sale',
        'is_mixable',
        'is_cocktail',

        'unit_capacity_type',
        'unit_capacity',
        'un_package',
    ];
    protected $hidden = [
        'created_at',
        'updated_at',
        'deleted_at'
    ];
    protected $casts = [
        'is_enable' => 'boolean',
        'can_change_price' => 'boolean',
        'is_cocktail' => 'boolean',
        'is_sale' => 'boolean',
        'is_mixable' => 'boolean',
    ];

    protected $appends = ['food_photo_url'];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function foodDishDetails()
    {
        return $this->hasMany(FoodDishDetail::class);
    }

    public function foodIngredients()
    {
        return $this->hasMany(FoodIngredient::class);
    }

    public function ingredients()
    {
        return $this->hasMany(FoodIngredient::class, 'ingredient_id');
    }

    // public function warehouseTransactions()
    // {
    //     return $this->hasMany(WarehouseTransaction::class, 'item_id');
    // }

    public function animalParts()
    {
        return $this->hasMany(AnimalPart::class);
    }

    public function saleDetails()
    {
        return $this->hasMany(SaleDetail::class);
    }

    public function saleDetailIngredients()
    {
        return $this->hasMany(SaleDetailIngredient::class, 'ingredient_id');
    }

    public function warehouse()
    {
        return $this->hasOne(Warehouse::class, 'item_id');
    }

    public function getFoodPhotoUrlAttribute()
    {
        if ($this->food_photo) {
            $url = Storage::url($this->food_photo);
            return $url;
            //return 'https://restaurant.khmernokor.com'. $url;
        }
        return null;
    }
}

File: xijinhui_window/XiJinHuiWindow/Models/FoodDish.cs
using XiJinHuiWindow.Utils;
using System;
using System.Drawing;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;
using XiJinHuiWindow.Properties;

namespace XiJinHuiWindow.Models
{
    public class FoodDish : HasNameLanguages
    {
        [JsonPropertyName("id")]
        public int? Id { get; set; }

        [JsonPropertyName("category_id")]
        public int? CategoryId { get; set; }
        public string CategoryName { get; set; }

        [JsonPropertyName("food_code")]
        public string FoodCode { get; set; }

        [JsonPropertyName("description")]
        public string Description { get; set; }

        [JsonPropertyName("food_price"), JsonConverter(typeof(StringToDoubleConverter))]
        public double? FoodPrice { get; set; }

        [JsonPropertyName("remark")]
        public string Remark { get; set; }

        [JsonPropertyName("food_photo")]
        public string FoodPhoto { get; set; }

        [JsonPropertyName("food_photo_url")]
        public string FoodPhotoUrl { get; set; }

        [JsonPropertyName("english_keyword")]
        public string EnglishKeyword { get; set; }

        [JsonPropertyName("chinese_keyword")]
        public string ChineseKeyword { get; set; }

        [JsonPropertyName("khmer_keyword")]
        public string KhmerKeyword { get; set; }

        [JsonPropertyName("display_order")]
        public int? DisplayOrder { get; set; }

        [JsonPropertyName("is_enable")]
        public bool? IsEnable { get; set; }

        [JsonPropertyName("can_change_price")]
        public bool? CanChangePrice { get; set; }

        [JsonPropertyName("deleted_at")]
        public string DeletedAt { get; set; }

        [JsonPropertyName("created_at")]
        public string CreatedAt { get; set; }

        [JsonPropertyName("updated_at")]
        public string UpdatedAt { get; set; }

        [JsonPropertyName("category")]
        public Category Category { get; set; }

        [JsonPropertyName("food_dish_details")]
        public List<FoodDishDetail> FoodDishDetails { get; set; }

        [JsonPropertyName("food_ingredients")]
        public List<FoodIngredient> FoodIngredients { get; set; }

        [JsonPropertyName("animal_parts")]
        public List<AnimalPart> AnimalParts { get; set; }
    }

}
